### PR TITLE
Data-check conversion from a previous Java-healthcheck

### PR DESCRIPTION
**Pull request by CristiGuijarro**

### CODE REVIEW

File: lib/Bio/EnsEMBL/DataCheck/Checks/CheckGenomicAlignCoverage.pm [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+  TABLES      => ['dnafrag', 'genome_db', 'genomic_align', 'method_link', 'method_link_species_set', 'method_link_species_set_tag', 'species_set_header', 'species_tree_node', 'species_tree_node_tag']
+};
+
+
+sub tests {
+  my ($self) = @_;
+    
+  my $overlap_check;
+  
+  if ($self->dba->dbc->dbname =~ /plants|pan/) {
```
Comment (muffato) [inline lib/Bio/EnsEMBL/DataCheck/Checks/CheckGenomicAlignCoverage.pm]: I would suggest to use the `get_division()` method that @JAlvarezJarreta introduced recently
```suggestion
  if ($self->dba->get_division() =~ /plants|pan/) {
```

File: lib/Bio/EnsEMBL/DataCheck/Checks/CheckGenomicAlignCoverage.pm [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+    #  1.1. in vertebrates:  reference species, but not human or mouse
+    #  1.2. non-vertebrates: overlaps allowed in all species
+    #2. sum of genomic_align ranges is larger than tag value when overlaps are allowed (non-ref species or ref human/mouse)
+    
+    my $tag_coverage_sql = qq/
+      SELECT LEFT(tag, 3) AS ref_status, GROUP_CONCAT(IF(tag LIKE '\%species', value, NULL)) AS species_name,
+        GROUP_CONCAT(IF(tag LIKE '\%coverage', value, NULL)) AS tag_coverage 
+      FROM method_link_species_set_tag 
+      WHERE (tag LIKE '\%species' OR tag LIKE '\%genome_coverage') 
+          AND method_link_species_set_id = $mlss_id GROUP BY LEFT(tag, 3)
```
Comment (muffato) [inline lib/Bio/EnsEMBL/DataCheck/Checks/CheckGenomicAlignCoverage.pm]: Can you move the `GROUP BY` to a new line ?

File: lib/Bio/EnsEMBL/DataCheck/Checks/CheckGenomicAlignCoverage.pm [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+    /;
+    
+    my $genomic_coverage_sql = qq/
+      SELECT g.name, d.genome_db_id, x.tag_coverage, SUM(ga.dnafrag_end-ga.dnafrag_start+1) AS genomic_align_coverage, 
+        $overlap_check AS overlaps_allowed 
+        FROM genomic_align ga JOIN dnafrag d USING(dnafrag_id) JOIN genome_db g USING(genome_db_id) 
+          JOIN ( $tag_coverage_sql ) x ON x.species_name = g.name 
+        WHERE ga.method_link_species_set_id = $mlss_id GROUP BY g.name
```
Comment (muffato) [inline lib/Bio/EnsEMBL/DataCheck/Checks/CheckGenomicAlignCoverage.pm]: Same here

File: lib/Bio/EnsEMBL/DataCheck/Checks/CheckGenomicAlignCoverage.pm [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+    /;
+  
+  my $msa_mlss_array = $helper->execute_simple(-SQL => $msa_mlss_sql);
+  foreach my $mlss_id (@$msa_mlss_array) {
+    
+    my $genomic_align_coverage_sql = qq/
+    SELECT d.genome_db_id, SUM(ga.dnafrag_end-ga.dnafrag_start+1) AS genomic_align_coverage 
+      FROM genomic_align ga 
+        JOIN dnafrag d USING(dnafrag_id) 
+      WHERE ga.method_link_species_set_id = $mlss_id GROUP BY d.genome_db_id
```
Comment (muffato) [inline lib/Bio/EnsEMBL/DataCheck/Checks/CheckGenomicAlignCoverage.pm]: And here

**Comment by CristiGuijarro** [comment]
Splitting into two different DCs as suggested

**Comment by CristiGuijarro** [comment]
Some of the other HC->DC conversions are better suited to some API calls (coming soon)

**Pull request merged**